locking/locks.go

```diff
@@ -22,31 +21,51 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
+type LockCacher interface {
+	Add(l Lock) error
+	RemoveByPath(filePath string) error
+	RemoveById(id string) error
+	Locks() []Lock
+	Clear()
+	Save() error
+}
+
 // Client is the main interface object for the locking package
 type Client struct {
-	cfg       *config.Configuration
-	apiClient *api.Client
-	cache     *LockCache
+	Remote string
+	client *lockClient
+	cache  LockCacher
 }
 
 // NewClient creates a new locking client with the given configuration
 // You must call the returned object's `Close` method when you are finished with
 // it
-func NewClient(cfg *config.Configuration) (*Client, error) {
-
-	apiClient := api.NewClient(api.NewHttpLifecycle(cfg))
+func NewClient(remote string, lfsClient *lfsapi.Client) (*Client, error) {
+	return &Client{
+		Remote: remote,
+		client: &lockClient{Client: lfsClient},
+		cache:  &nilLockCacher{},
+	}, nil
+}
 
-	lockDir := filepath.Join(config.LocalGitStorageDir, "lfs")
-	err := os.MkdirAll(lockDir, 0755)
+func (c *Client) SetupFileCache(path string) error {
```

Comment:
Option funcs are good if the values are needed in the constructor. Otherwise, setting properties seems to be closer to go conventions. You only have to call `SetupFileCache()` if you want a cacher.